cmd/drand-gossip-relay/lp2p/ctor.go

```diff
@@ -0,0 +1,82 @@
+package lp2p
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+
+	"github.com/ipfs/go-datastore"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/host"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	routing "github.com/libp2p/go-libp2p-routing"
+	libp2ptls "github.com/libp2p/go-libp2p-tls"
+	xerrors "golang.org/x/xerrors"
+)
+
+var (
+	log       = logging.Logger("lp2p")
+	privDsKey = datastore.NewKey("p2pPrivKey")
+)
+
+func PubSubTopic(nn string) string {
+	return fmt.Sprintf("/drand/pubsub/v0.0.0/%s", nn)
+}
+
+func ConstructHost(priv crypto.PrivKey, listenAddr string) (host.Host, *dht.IpfsDHT, error) {
+	var idht *dht.IpfsDHT
+
+	ctx := context.TODO()
+	h, err := libp2p.New(ctx,
+		libp2p.ListenAddrStrings(listenAddr),
+		libp2p.Identity(priv),
+		libp2p.Security(libp2ptls.ID, libp2ptls.New),
+		libp2p.Routing(func(h host.Host) (routing.PeerRouting, error) {
+			var err error
+			idht, err = dht.New(ctx, h)
+			return idht, err
+		}),
+		libp2p.DisableRelay(),
+	)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("constructing host: %w", err)
+	}
+	return h, idht, nil
+}
+
+func ConstructPubSub(h host.Host) (*pubsub.PubSub, error) {
+	p, err := pubsub.NewGossipSub(context.TODO(), h)
```

Comment:
How do two separate instances of drand-gossipsub-relay discover each other?

Reply:
They have to connect to each other, in whatever topology. I will add `connect-with` file.

Reply:
Added `peer-with` option. Push incoming.